shared/ssl.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+#
+# Get a signed certificate for this host.
+#
+# Uses OpenSSL and curl directly to generate a new CSR and get it signed by the
+# Puppet Server CA.
+#
+# Intended to be used in place of a full-blown puppet agent run that is solely
+# for getting SSL certificates onto the host.
+#
+# Files will be placed in the same default directory location and structure
+# that the puppet agent would put them, which is /etc/puppetlabs/puppet/ssl.
+#
+# Fails under the following conditions:
+# - CA host cannot be reached
+# - CA already has signed certificate for this host
+# - SSL files already exist on this host
+#
+# Arguments:
+#   $1  (Optional) Certname to use, defaults to $HOSTNAME
+#
+# Optional environment variables:
+#   WAITFORCERT            Number of seconds to wait for certificate to be
+#                          signed, defaults to 120
+#   PUPPETSERVER_HOSTNAME  Hostname of Puppet Server CA, defaults to "puppet"
+#   SSLDIR                 Root directory to write files to, defaults to
+#                          "/etc/puppetlabs/puppet/ssl"
+
+msg() {
+    echo "($0) $1"
+}
+
+error() {
+    msg "Error: $1"
+    exit 1
+}
+
+### Verify dependencies available
+! which openssl > /dev/null && error "openssl not found on PATH"
+! which curl > /dev/null && error "curl not found on PATH"
+
+### Verify options are valid
+CERTNAME="${1:-${HOSTNAME}}"
+[ -z "${CERTNAME}" ] && error "certificate name must be non-empty value"
+PUPPETSERVER_HOSTNAME="${PUPPETSERVER_HOSTNAME:-puppet}"
+[ -z "${PUPPETSERVER_HOSTNAME}" ] && error "PUPPETSERVER_HOSTNAME must be non-empty value"
```

Comment:
`PUPPETSERVER_HOSTNAME` can't be empty at this point.

Reply:
You're saying this check is not needed, and we can remove it?

Or that we should change this check?

Reply:
Sorry for being obtuse: yes, I meant that the check should be removed since it can't fail.

Reply:
I wasn't sure if there was a way to specify `""` as a value by accident, especially if this script is called from another script or something, so was just trying to be real defensive here.

Reply:
I like removing unnecessary things :)

Removing these 2 checks....